src/model/AIState.hpp

```diff
@@ -11,58 +11,206 @@
+                                      const spy::gameplay::State &s);
+
+            void processGadgetNugget(std::shared_ptr<const spy::gameplay::GadgetAction> action,
+                                              const spy::gameplay::State &s,
+                                              spy::character::FactionEnum me);
+
+            void processGadgetBowlerBlade(std::shared_ptr<const spy::gameplay::GadgetAction> action,
+                                                   const spy::gameplay::State &s, const spy::MatchConfig &config);
+
+            void processGadgetCocktail(std::shared_ptr<const spy::gameplay::GadgetAction> action,
+                                                const spy::gameplay::State &s);
+
+            void processGadgetMoledie(std::shared_ptr<const spy::gameplay::GadgetAction> action,
+                                               const spy::gameplay::State &s, const spy::MatchConfig &config);
+
+            void processGadgetLaserCompact(std::shared_ptr<const spy::gameplay::GadgetAction> action,
+                                                    const spy::gameplay::State &s);
+
+            void processGadgetPoisonPills(std::shared_ptr<const spy::gameplay::GadgetAction> action,
+                                                   const spy::gameplay::State &s);
+
+            void processGadgetChickenFeed(std::shared_ptr<const spy::gameplay::GadgetAction> action,
+                                                   const spy::gameplay::State &s);
+
+            void processGadgetMirrorOfWilderness(std::shared_ptr<const spy::gameplay::GadgetAction> action,
+                                                          const spy::gameplay::State &s);
+
+            void processGadgetHairdryer(std::shared_ptr<const spy::gameplay::GadgetAction> action,
+                                                 const spy::gameplay::State &s);
+
+            void processGettingRidOfMoledie(std::shared_ptr<const spy::gameplay::BaseOperation> operation);
+
+            /**
+             * get faction of character according to AIState
+             * @param id character id of the character faction wants to be known
+             * @return faction, INVALID if unknown, NEURAL if npc, PLAYER1 if my, PLAYER2 if enemy
+             */
+            spy::character::FactionEnum getFaction(const spy::util::UUID &id);
+
+            /**
+             * find out if character has certain property
+             * @param id id of the character to be searched for
+             * @param prop property to be checked
+             * @return true if character has property, else false
+             */
+            bool hasCharacterProperty(const spy::util::UUID &id, spy::character::PropertyEnum prop);
+
+            void modifyUsagesLeft(const std::shared_ptr<spy::gadget::Gadget> &gad);
+
+            void push_back_toUnknownGadgets(std::shared_ptr<spy::gadget::Gadget> key, const spy::util::UUID &val,
```

Comment:
Warum hier die Mischung aus snake und camel case?

Reply:
Weil in der std lib das auch push_back heißt und das quasi nur eine custom Implementierung davon ist.